docs/data/date-pickers/lifecycle/lifecycle.md

```diff
@@ -0,0 +1,333 @@
+---
+product: date-pickers
+title: Components lifecycle
+githubLabel: 'component: pickers'
+packageName: '@mui/x-date-pickers'
+---
+
+# Components lifecycle
+
+<p class="description">This page explains when the callbacks onChange, onAccept and onClose are called.</p>
+
+## Fields lifecycle
+
+### On simple fields
+
+:::info
+The information below are applicable on standalone fields (when rendering `<DateField />`),
+but also on pickers for the field editing (when rendering `<DatePicker />` and using the input to edit the value).
+:::
+
+The field components have an internal state to update the visible value.
+
+It will only call the `onChange` callback when:
+
+- the user fills one section of an empty input. The value equals `Invalid date`.
+- the user completes all sections of an input. The value reflects the input.
+- the user cleans one section of a completed input. The value equals `Invalid date`.
+- the user cleans all sections of an input. The value equals `null`.
+
+In the example below, `onChange` will be called when any of the conditions are triggered:
+
+{{"demo": "LifeCycleDateFieldEmpty.js", "defaultCodeOpen": false}}
+
+### On range fields [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
+
+On range fields (`SingleInputDateRangeField` / `MultiInputDateRangeField` / ... ),
+`onChange` will be called if the date you are modifying is matching one of the condition above,
+even if the other date does not.
+
+In the example below, changing the value of the start date section will call `onChange` even if the end date is empty or partially filled.
+
+{{"demo": "LifeCycleDateRangeField.js", "defaultCodeOpen": false}}
+
+## Pickers lifecycle
+
+### When is `onClose` called ?
+
+:::info
+If you are not controlling the `open` prop, then all the scenarios below also describe when the picker closes.
+:::
+
```

Comment:
It's confusing. I mean it's expected that If I don't control the open/close behavior the `onClose` is called when closing. This info let me thought "Ok, if it's highlighted their might be a trick" But did not find it.

```suggestion

```

Reply:
How would you phrase it?
The information I want to cover here is that the section covers explains when the picker is closed.
But if you control the open status, then it covers when `onClose` is called.

Maybe I can rename the section `When is it closed?`

Reply:
Just with the current title `### When is `onClose` called?` it sounds good.

It's not that much of an issue that by default when `onClose` is called, the view closes.

Reply:
`onClose` is called even when you are not controlling.

Reply:
Did I get this right? We can split the information in smaller sentences. 🤔 

```suggestion
:::info
In all the below scenarios the picker closes by default. Additionally, `onClose` is called in these scenarios if the `open` prop is controlled.
:::
```

Reply:
> Additionally, `onClose` is called in these scenarios if the `open` prop is controlled.

`onClose` is called even if you are not controlling

> In all the below scenarios the picker closes by default

I think it's hard to guess what "by default" means.

How about

:::info
In all the below scenarios, the picker closes when `onClose` is called, except if you are controlling the `open` prop..
:::